### ISSUE TITLE
Add svg dimensions to sprite object

### ISSUE THREAD

**Issue by galymax** [cascornelissen/svg-spritemap-webpack-plugin#37]
When using the svg sprite in sass as background, the container dimensions or background size need to be used to scale the svg. Often the svg needs to be used in ins original dimensions, so it would be really helpful to be able to access these in sass. Could you consider changing the generated sprite object like so:

```
$sprites: (
    'mySVG': (
	svg: "data:image/svg+xml,%3csvg xmlns...",
	width: 14.5,
	height: 5
   )
);
```

The mixin would need to be changed like so:

```
@mixin sprite($name) {
    $obj: map-get($sprites, $name);
    $sprite: map-get($obj, svg);
    background: url($sprite) center no-repeat;
}
```

That would add the possibility to overwrite the mixin with a custom one that uses the original dimensions of the svg:

```
@mixin sprite($name) {
    $obj: map-get($sprites, $name);
    $sprite: map-get($obj, svg);
    background: url($sprite) center no-repeat;
    background-size: map-get($obj, width)px map-get($obj, height)px;
}
```

You could also consider adding an optional parameter to the mixin that sets the svg to it's original size.

Thanks!

**Comment by robertklein182** [comment]
These changes should help
```js
// generate-svg.js L70

symbol.setAttribute("width", width);
symbol.setAttribute("height", height);
```


```js
// style-formatters/scss-formater.js L34

const width = symbol.getAttribute("width");
const height = symbol.getAttribute("height");

return `"${selector}": ("width": ${width}, "height": ${height}, "sprite": "${dataURI}")`;
```

```scss
// templates/styles.scss L5

@mixin sprite($name, $width: auto, $height: auto) {
    $values: map-get($sprites, $name);
    $sprite: map-get($values, "sprite");

    $svgWidth: map-get($values, "width");
    $svgHeight: map-get($values, "height");

    $newHeight: $svgHeight / $svgWidth;

    display: inline-block;
    background: url($sprite) center no-repeat;

    @if $width != auto and $height == auto {
        $svgWidth: $width;
        $newHeight: $newHeight * $width;
        width: #{$width}px;
        height: #{$newHeight}px;
    }
    @if $width != auto and $height != auto {
        width: #{$width}px;
        height: #{$height}px;
    }
    @if $width == auto and $height == auto {
        width: #{$svgWidth}px;
        height: #{$svgHeight}px;
    }
}
```


I don't know if it makes sense to set the height parameter, because we shouldn't change the ratio of an image. I also had a problem with setting `$newHeight` inside the if-clause.

**Comment by cascornelissen** [comment]
Thanks for the feature request, sounds like a great thing to add. The only problem I'm seeing is that the solution provided by @galymax (rewriting the `$sprites` variable) would be backwards incompatible.

We could either think of a different setup, maybe involving a separate `$sizes` variable or holding on to this for a `3.x` release which I'm planning either way as I would love to refactor the current options setup.

**Comment by galymax** [comment]
Yeah, I also thought about the backwards compatibility issue, as apparently some people are accessing the sprites directly in their setup. I think a separate sizes variable is fine, as long as the sizes are accessed through the same icon names as the svg data, that wouldn't complicate things.
```
$sprites: (
    'mySVG': 'data:image/svg+xml,%3csvg xmlns...'
);
$sizes: (
    'mySVG': (
        'width': 14.5,
        'height': 5
    )
);
```
The usage in the mixin stays practically the same as in my first example:
```
@mixin sprite($name) {
    $sprite: map-get($sprites, $name);
    $size: map-get($sizes, $name);
    background: url($sprite) center no-repeat;
    background-size: map-get($size, width)px map-get($size, height)px;
}
```

**Comment by cascornelissen** [comment]
Agreed, if you want you're free to submit a PR, if not I'll take a look at this when I have some more time.

**Comment by cascornelissen** [comment]
@galymax, thanks again for the suggestion and PR. This feature has been released in version [`2.7.0`](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/releases/tag/v2.7.0). 🚀
